Include/internal/pycore_optimizer.h

```diff
@@ -13,6 +13,16 @@ extern "C" {
 int _Py_uop_analyze_and_optimize(PyCodeObject *code,
     _PyUOpInstruction *trace, int trace_len, int curr_stackentries);
 
+extern PyTypeObject _PyCounterExecutor_Type;
+#define _PyCounterExecutor_Check(op) Py_IS_TYPE((op), &_PyCounterExecutor_Type)
+extern PyTypeObject _PyCounterOptimizer_Type;
+#define _PyCounterOptimizer_Check(op) Py_IS_TYPE((op), &_PyCounterOptimizer_Type)
+extern PyTypeObject _PyDefaultOptimizer_Type;
+#define _PyDefaultOptimizer_Check(op) Py_IS_TYPE((op), &_PyDefaultOptimizer_Type)
+extern PyTypeObject _PyUOpExecutor_Type;
+#define _PyUOpExecutor_Check(op) Py_IS_TYPE((op), &_PyUOpExecutor_Type)
+extern PyTypeObject _PyUOpOptimizer_Type;
+#define _PyUOpOptimizer_Check(op) Py_IS_TYPE((op), &_PyUOpOptimizer_Type)
```

Comment:
We don't need these type check macros for this change, and they can always be added at a later date if that changes:

```suggestion
extern PyTypeObject _PyCounterOptimizer_Type;
extern PyTypeObject _PyDefaultOptimizer_Type;
extern PyTypeObject _PyUOpExecutor_Type;
extern PyTypeObject _PyUOpOptimizer_Type;
```